### PR TITLE
basic GitHub Actions; fix tests; GHC 9 still failing

### PR DESCRIPTION
**Pull request by peterbecich**
https://github.com/peterbecich/servant-ekg/actions/runs/1738605018

### CODE REVIEW

**Comment by peterbecich** [comment]
The tests are failing to build, but https://github.com/haskell-servant/servant-ekg/pull/21 might help

**Comment by arianvp** [comment]
One of the issues is simply a missing instance for `Fragment a :> api`. Which should be the same as the `QueryParam a` instance. Fixes #29

**Comment by peterbecich** [comment]
To fix building with GHC 9, there are some dependencies which may need to be updated.

For GHC 8, everything seems to be good now: https://github.com/peterbecich/servant-ekg/runs/4994048104?check_suite_focus=true

**Comment by arianvp** [comment]
Alright; shall we turn GHC 9 support into a new PR and merge tthis as is?

**Review by arianvp — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      fail-fast: false
+      matrix:
+        versions:
+          - ghc: '8.6.5'
+            cabal: '3.6'
+          - ghc: '8.8.4'
+            cabal: '3.6'
+          - ghc: '8.10.7'
+            cabal: '3.6'
+          - ghc: '9.0.2'
```
Comment (arianvp) [inline .github/workflows/ci.yml]: we could comment out 9.0.2 for now

Comment (arianvp) [reply .github/workflows/ci.yml]: We should update the `tested-with` list in the .Cabal file ideally

**Review by arianvp — APPROVED** [review]

**Pull request merged**